tuned/plugins/plugin_video.py

```diff
@@ -43,49 +42,83 @@ def _radeon_powersave_files(self, device):
 			"profile": "/sys/class/drm/%s/device/power_profile" % device,
-			"dpm_state": "/sys/class/drm/%s/device/power_dpm_state" % device
+			"dpm_state": "/sys/class/drm/%s/device/power_dpm_state" % device,
+			"level": "/sys/class/drm/%s/device/power_dpm_force_performance_level" % device
 		}
 
 	@command_set("radeon_powersave", per_device=True)
 	def _set_radeon_powersave(self, value, device, sim):
-		sys_files = self._radeon_powersave_files(device)
-		va = str(re.sub(r"(\s*:\s*)|(\s+)|(\s*;\s*)|(\s*,\s*)", " ", value)).split()
-		if not os.path.exists(sys_files["method"]):
-			if not sim:
-				log.warn("radeon_powersave is not supported on '%s'" % device)
+		if not sim:
+			device_obj = self._get_device_objects((device,))
+			module = None
+			if device_obj:
+				module = self._device_module_name(device_obj[0])
+			if module not in ["amdgpu", "radeon"]:
+				log.warn("radeon_powersave is not supported on '%s' (module '%s')" % (device, module))
 				return None
-		for v in va:
-			if v in ["default", "auto", "low", "mid", "high"]:
-				if not sim:
-					if (self._cmd.write_to_file(sys_files["method"], "profile") and
+
+			sys_files = self._radeon_powersave_files(device)
+			va = str(re.sub(r"(\s*:\s*)|(\s+)|(\s*;\s*)|(\s*,\s*)", " ", value)).split()
+			for v in va:
+				if v in ["default", "auto", "low", "mid", "high"]:
+					if module == "amdgpu":
+						log.warn("Option '%s' not supported on amdgpu module." % v)
+					# https://wiki.archlinux.org/index.php/ATI#Profile-based_frequency_switching
+					elif (self._cmd.write_to_file(sys_files["method"], "profile") and
 						self._cmd.write_to_file(sys_files["profile"], v)):
 						return v
-			elif v == "dynpm":
-				if not sim:
-					if (self._cmd.write_to_file(sys_files["method"], "dynpm")):
+				elif v == "dynpm":
+					if module == "amdgpu":
+						log.warn("Option '%s' not supported on amdgpu module." % v)
+					# https://wiki.archlinux.org/index.php/ATI#Dynamic_frequency_switching
+					elif self._cmd.write_to_file(sys_files["method"], "dynpm"):
 						return "dynpm"
-			# new DPM profiles, recommended to use if supported
-			elif v in ["dpm-battery", "dpm-balanced", "dpm-performance"]:
-				if not sim:
+				elif v in ["dpm-battery", "dpm-balanced", "dpm-performance"]:
 					state = v[len("dpm-"):]
-					if (self._cmd.write_to_file(sys_files["method"], "dpm") and
+					# https://www.kernel.org/doc/html/latest/gpu/amdgpu.html#power-dpm-state
+					if module == "amdgpu":
+						log.warn("Option '%s' not supported on amdgpu module." % v)
+					# https://wiki.archlinux.org/index.php/ATI#Dynamic_power_management
+					elif (self._cmd.write_to_file(sys_files["method"], "dpm") and
 						self._cmd.write_to_file(sys_files["dpm_state"], state)):
 						return v
-			else:
-				if not sim:
-					log.warn("Invalid option for radeon_powersave.")
-				return None
+				# new settings for amdgpu module
+				elif v in ["level-auto", "level-low", "level-high"]:
+					level = v[len("level-")]
```

Comment:
It seems you write just the first char after the 'level-', e.g. 'a', not 'auto' for 'level-auto', I think it needs `v[len("level-"):]`.